MaxText/layers/attentions.py

```diff
@@ -353,10 +360,12 @@ def qk_product(self, query: Array, key: Array) -> Array:
     n_kv = key.shape[-2]
     assert n_kv == self.num_kv_heads
     query = jnp.reshape(query, (b, t, n_kv, n // n_kv, d))
+    if self.reshape_q and q_seq_len == 1:
+      query = jnp.broadcast_to(query, (b, 2, n_kv, n // n_kv, d))
```

Comment:
Just out of my personal curiosity. It seems we are trying to broadcast the t axis (always be 1 for decode phase) to 2 to force the MXU to performance the matmul instead of on VPU.

Is this also doable by removing the t axis by `jnp.squeeze` which might be better than broadcast to save the computation and memory cost?